### DISCUSSION TITLE
5.2.2 Released

### DISCUSSION THREAD

**Discussion by sbwalker** [oqtane/oqtane.framework#4652] [General]
https://www.oqtane.org/blog/!/99/oqtane-5-2-2-released

Changes in v5.2.2

https://github.com/oqtane/oqtane.framework/pull/4645 Add Logout Everywhere option to User Profile @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4645 Logged IP address during user login @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4645 Fixed issue in Log Manager which could result in index out of range exception @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4643 Highlighted default site theme in list of themes displayed in page management @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4640 Preserved current Url for all Module Actions @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4637 Synchronize interop.js with .NET MAUI @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4636 Improved LoadJS logic in interop.js to handle scripts already registered @zyhfish
https://github.com/oqtane/oqtane.framework/pull/4635 Implemented deep cloning to not mutate cache @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4633 Fixed trimming of site, page, and module settings @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4632 Added ability to load resources for Theme and Container setting components @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4631 Added message in Blazor Hybrid scenario if Security Token has not been configured for file upload @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4630 Synchronized static resources with .NET MAUI solution @zyhfish
https://github.com/oqtane/oqtane.framework/pull/4629 Preserved ReturnUrl when navigating to Settings component @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4627 Added FileLogger as fallback in LogManager when site cannot be determined @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4626 Fixed logic to force authenticated users to provide email address in static rendering @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4625 Updated nuget.exe to latest and fix nuspec specification of readme @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4623 Move principal creation in external login to occur after role processing logic @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4622 Allowed external login to support SecurityStamp @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4620 Allowed JwtMiddleware to support SecurityStamp @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4619 Enabled SecurityStamp in IdentityRevalidatingAuthenticationStateProvider @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4618 Added support for "logout everywhere" using SecurityStamp (credit @marcdrexel) @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4617 Improved script injection in static rendering @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4616 Include SecurityStamp in cached User object @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4613 Improved support for site name in logo component @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4612 Allowed logo to show site name as fallback (credit @JanOlsmar) @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4611 Fixed Site HeadContent scripts being added twice @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4609 Improved support for external login roles @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4608 Optimized logic in toggle edit mode @hishamco
https://github.com/oqtane/oqtane.framework/pull/4606 Improved user experience for file upload @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4605 Relocated logic to filter user settings in API layer @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4601 Avoided null exception in Module Settings @hishamco
https://github.com/oqtane/oqtane.framework/pull/4597 Ensured all state is initialized before rendering permission grid @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4596 Added support for DateOnly and TimeOnly columns in migrations @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4593 Fixed IconOnly property behavior in ActionDialog @mdmontesinos
https://github.com/oqtane/oqtane.framework/pull/4588 Updated MySQL Project File version to 5.2.2 @thabaum
https://github.com/oqtane/oqtane.framework/pull/4582 Removed reference to HttpContext from Search component as it is not used @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4578 Fixed issue adding existing user to a new site @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4577 Abstracted Bootstrap references so they can be updated in a single location @zyhfish
https://github.com/oqtane/oqtane.framework/pull/4576 Improved developer experience for Url helper methods @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4574 Improved upgrade user experience to display the upgrade progress @zyhfish
https://github.com/oqtane/oqtane.framework/pull/4573 Filtered deleted modules from Control Panel add existing and copy existing options @leigh-pointer
https://github.com/oqtane/oqtane.framework/pull/4571 Provided better support for AllowTextInput on Search component @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4570 Fixed progress indicator in search results in static rendering @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4568 Prevented Log fields from exceeding column length @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4566 Fixed Search component so that it checks if search is enabled for site, and include AllowTextInput parameter to control input textbox @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4565 Fixed default module template issue caused by gitignore @sbwalker
https://github.com/oqtane/oqtane.framework/pull/4555 Add readme.md to Oqtane nuget packages @sbwalker

List of Contributors in v5.2.2

@sbwalker
@leigh-pointer
@thabaum
@zyhfish
@mdmontesinos
@hishamco
@JanOlsmar
@marcdrexel

**Comment by thabaum** [comment]
Successfully upgraded IIS production site from 5.2.1 to 5.2.2 with blog module installed without error in Event Log.

However I did see these in SystemInfo > Log

[2024-09-23 18:05:05+00:00] [Error] An exception was thrown while deserializing the token.    at Microsoft.AspNetCore.Antiforgery.DefaultAntiforgeryTokenSerializer.Deserialize(String serializedToken)
   at Microsoft.AspNetCore.Antiforgery.DefaultAntiforgery.GetCookieTokenDoesNotThrow(HttpContext httpContext)
[2024-09-23 18:15:18+00:00] [Error] An exception was thrown while deserializing the token.    at Microsoft.AspNetCore.Antiforgery.DefaultAntiforgeryTokenSerializer.Deserialize(String serializedToken)
   at Microsoft.AspNetCore.Antiforgery.DefaultAntiforgery.GetCookieTokenDoesNotThrow(HttpContext httpContext)
[2024-09-23 18:15:18+00:00] [Error] An exception was thrown while deserializing the token.    at Microsoft.AspNetCore.Antiforgery.DefaultAntiforgeryTokenSerializer.Deserialize(String serializedToken)
   at Microsoft.AspNetCore.Antiforgery.DefaultAntiforgery.GetCookieTokenDoesNotThrow(HttpContext httpContext)
   
   I also see i had my log options set to "Information"
   
   I will test on another instance when I have time with full details.

**Reply by markdav-is** [reply]
I'm getting this error today updating a site on smarterasp.net to oqtane6.  @thabaum what was your fix?

**Comment by thabaum** [comment]
@sbwalker Any speed bench tests we report prior to .NET 9?  Maybe we can keep a thread going to showcase the improvements as we enhance the framework along with Microsoft .NET Core Blazor/Maui improvements?

**Comment by thabaum** [comment]
Log out everywhere seems to be working great!  Thank you @sbwalker for this security enhancement in 5.2.2!

**Comment by sbwalker** [comment]
@thabaum if you are volunteering to capture some performance metrics I am sure others would appreciate it

**Comment by thabaum** [comment]
@sbwalker I will if you can give them to me for my site, and others can as well share I will create a thread, thanks!

On another note, you see the time +00.00

Current system time is : 9/23/2024 6:32:37 PM UTC

Where would we correct this?

**Comment by thabaum** [comment]
I meant to say give me some sites we should use for solid metrics that is reliable to share.

**Comment by sbwalker** [comment]
@thabaum 

"On another note, you see the time +00.00

Current system time is : 9/23/2024 6:32:37 PM UTC"

I have no idea what you are talking about

**Reply by thabaum** [reply]
![image](https://github.com/user-attachments/assets/139f2311-a509-4f1e-b6eb-79c6380f4b8e)

**Reply by sbwalker** [reply]
9/23/2024 6:32:37 PM UTC is the exact same as 9/23/2024 18:32:37+00:00

**Reply by thabaum** [reply]
Admin Dashboard > System Info

we should be able to set the site time zone?  Enhancement request possibly, I just want to be sure before making it that it is not already there and I just didnt find it yet.

**Reply by thabaum** [reply]
It is 11:40AM here (edited) -7 we need a way to set this so logs look at the time of the administrator I believe if you follow me.

**Reply by sbwalker** [reply]
All dates in Oqtane are stored in UTC time.

**Reply by thabaum** [reply]
where does it get the time?  From the server or the web?

**Reply by sbwalker** [reply]
from the server

**Reply by thabaum** [reply]
OK I will review the server time, which gets it from the web, should be up-to-date.  

What I guess I am saying is the timezone needs set in Oqtane.

Possibly here?

![image](https://github.com/user-attachments/assets/5d4b5664-61e7-4d54-a99b-83626050dcc3)

**Reply by sbwalker** [reply]
I am not interested in adding this capability at this time
